src/datasets/load.py

```diff
@@ -658,29 +755,33 @@ def get_module(self) -> DatasetModule:
         builder_kwargs = {
             "hash": hash,
             "data_files": data_files,
-            "config_name": os.path.basename(self.path.rstrip("/")),
+            "config_name": self.config_name,
             "base_path": self.path,
             **builder_kwargs,
         }
         if os.path.isfile(os.path.join(self.path, config.DATASETDICT_INFOS_FILENAME)):
             with open(os.path.join(self.path, config.DATASETDICT_INFOS_FILENAME), encoding="utf-8") as f:
                 dataset_infos: DatasetInfosDict = json.load(f)
-            if dataset_infos:
-                builder_kwargs["config_name"] = next(iter(dataset_infos))
-                builder_kwargs["info"] = DatasetInfo.from_dict(next(iter(dataset_infos.values())))
-        if os.path.isfile(os.path.join(self.path, "README.md")):
-            dataset_metadata = DatasetMetadata.from_readme(Path(self.path) / "README.md")
-            if isinstance(dataset_metadata.get("dataset_info"), list) and dataset_metadata["dataset_info"]:
-                dataset_info_dict = dataset_metadata["dataset_info"][0]
+            dataset_info = dataset_infos.get(self.config_name, None)
+            if dataset_info:
+                builder_kwargs["info"] = DatasetInfo.from_dict(dataset_info)
```

Comment:
If there is only one config, shouldn't it pick it ?

For example `squad` has one config "plain_text". It is the default config because it's the only one.

Reply:
makes sense !